src/AWS/CRT/Options.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace AWS\CRT;
+
+final class Options {
+    private $options = [];
+
+    public function __construct($opts = []) {
+        $this->options = $opts;
+    }
+
+    public function __get($name) {
+        return (isset($this->options[$name])) ? $this->options[$name] : null;
+    }
+
+    public function asArray() {
+        return $this->options;
+    }
+
+    public function toArray() {
+        return array_merge_recursive([], $this->options);
+    }
+
+    public function getInt($name) {
+        $val = $this->$name;
```

Comment:
similar to below.  Also, I think you meant $this->options[$name], not $this->$name

Reply:
Nope, this is actually correct, this will invoke` __get($name)`

Reply:
Ahh magic methods, I forgot about this one